toplevel/toploop.ml

```diff
@@ -105,10 +119,13 @@ let run_script ppf name args =
   Sys.interactive := false;
   run_hooks After_setup;
   let explicit_name =
+    match name with
+    | File name as filename  -> (
     (* Prevent use_silently from searching in the path. *)
     if name <> "" && Filename.is_implicit name
-    then Filename.concat Filename.current_dir_name name
-    else name
+    then File (Filename.concat Filename.current_dir_name name)
+    else filename)
+    | x -> x
```

Comment:
Could you list the two other cases explicitly? (Could be just `(String _ | Stdin) as x`.) This will force us to revisit this pattern-matching if we add a new source of input.

Reply:
changed